design-proposals/nic-hotplug/nic-hotplug.md

```diff
@@ -472,14 +392,14 @@ scenarios:
 
 # Implementation Phases
 1. **M** Refactor multus as a thick cni plugin.
-2. **M** Refactor multus to allow adding / removing only new networks
-3. **C** Consume this updated multus functionality via CNAO
-4. **K** Add the hot-plug / hot-unplug functionality to KubeVirt
-5. **M** Take into account the existing interfaces when multus implicitly generates
-   new interface names
-6. **M** update the default route on the pod whenever the NIC holding the
-   default route is hot-unplugged. It should now point to the cluster's default
-   network - managed by Kubernetes.
+2. **M** Expose an extra endpoint in multus to allow adding / removing specific
+   attachments
+3. Add a controller monitoring pod attachment updates
+4. **C** Consume this dynamic networks functionality via CNAO
+5. **K** Add the hot-plug functionality to KubeVirt for L2 networks
+6. **K** Extend the hot-plug functionality to KubeVirt for L3 networks (handle
+   IPAM)
+7. **K** Add the hot-unplug functionality to KubeVirt for L2 and L3 networks
```

Comment:
I think unplugging L2 can allow the feature to be complete/GA.

While the L3 part is an optional extension, especially when we know that it introduces some issues which are harder to handle.

Reply:
Yes, exactly.